### DISCUSSION TITLE
Ideas & Suggestions

### DISCUSSION THREAD

**Discussion by asiryan** [asiryan/FaceONNX#30] [Ideas]
Dear users and colleagues, if you are interested in [FaceONNX](https://github.com/FaceONNX/FaceONNX) or its derivative libraries, please share your ideas and suggestions. I would love to discuss them with you.

**Comment by yangjieshao** [comment]
是否可以对证件照做相关优化
Can we optimize for ID photo

证件照有以下问题
The ID photo has the following issues

1. 照片尺寸比较小
The photo size is relatively small

2. 照片拍摄年份可能比较久远
The photo may have been taken a long time ago
比如有的照片是20岁拍摄的
For example, some photos were taken at the age of 20
但是本人现在已经50岁了
But the person is now 50 years old

3. 拍摄时的由于当时的技术原因，照片比较糊
Due to technical reasons at the time of shooting, the photos were quite blurry

使用时，可以标识出哪张是证件照哪张是现场拍摄的照片
When in use, it can be identified which photo is the identification photo and which photo was taken on-site
在未优化前，拍摄时间比较久远的证件照基本无法通过对比判断是否是同一个人
Before optimization, it was almost impossible to determine whether ID photos taken for a long time were the same person through comparison

当然在需要说明的是
Of course, it should be noted that
正式使用的时候，证件照对比只会用在1:1对比场景而不是1:n场景
When officially used, ID photo comparison will only be used in 1:1 comparison scenarios instead of 1: n scenarios

**Comment by william-G2hOux9g8q** [comment]
It would be nice to be able to load the models from external sources instead of having them embedded (HTTP URL, UNC path, etc.).
Support for some of the other gender/age models would also be nice (antelopev2, etc.), but I'm not sure how difficult it would be to detect and run inference for the different models.

edit: Actually, instead of anything specific like HTTP loading it would probably be best to just have a constructor that takes a byte array or a stream, since that would make it easy to load from arbitrary sources.
